projects/igniteui-angular/schematics/utils/dependency-handler.ts

```diff
@@ -156,24 +156,21 @@ const addHammerToConfig =
 const includeDependencies = async (pkgJson: any, context: SchematicContext, tree: Tree): Promise<void> => {
     const workspaceHost = createHost(tree);
     const { workspace } = await workspaces.readWorkspace(tree.root.path, workspaceHost);
-    const defaultProject = getDefaultProjectFromWorkspace(workspace);
+    const projects = getProjectsFromWorkspace(workspace);
+
     for (const pkg of Object.keys(pkgJson.dependencies)) {
         const version = pkgJson.dependencies[pkg];
         const entry = DEPENDENCIES_MAP.find(e => e.name === pkg);
         if (!entry || entry.target === PackageTarget.NONE) {
             continue;
         }
-        switch (pkg) {
-            case 'hammerjs':
-                logIncludingDependency(context, pkg, version);
-                addPackageToPkgJson(tree, pkg, version, entry.target);
-                await addHammerToConfig(defaultProject, tree, 'build', context);
-                await addHammerToConfig(defaultProject, tree, 'test', context);
-                break;
-            default:
-                logIncludingDependency(context, pkg, version);
-                addPackageToPkgJson(tree, pkg, version, entry.target);
-                break;
+        logIncludingDependency(context, pkg, version);
+        addPackageToPkgJson(tree, pkg, version, entry.target);
+        if (pkg === 'hammerjs') {
+            for (let project of projects.values()) {
```

Comment:
I cannot, because of the await, few lines below.

Reply:
```typescript
projects.forEach(async (project) => {
	await addHammerToConfig(project, tree, 'build', context);
	await addHammerToConfig(project, tree, 'test', context);
});
```

Reply:
@kdinev I remember having the failing test when using `forEach` instead of `for of` and after going through some sources (like [this](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop)) I decided to stick with `for of`

Reply:
Use `Promise.all(projects.map(...))` then. I don't see why this has to be executed in sequence.

Reply:
workspace.projects type is not an `array`, but `ProjectDefinitionCollection`, which implements `ReadonlyMap<string, V>` . That's why the following code is needed:

```
            await Promise.all(Array.from(workspace.projects.values()).map(async (project) => {
                await addHammerToConfig(project, tree, 'build', context);
                await addHammerToConfig(project, tree, 'test', context);
            }));
```

I guess it is still faster than writing files in sequence, but as a code is not better than `for of`.

Reply:
@Lipata This is I/O code. I don't see how it executing more efficiently makes it worse than a more inefficient implementation?

Reply:
Done